### DISCUSSION TITLE
[API] Metadata of other type then string are not visible under "Additional information" section on the invoice page of Btcpay

### DISCUSSION THREAD

**Discussion by learntheropes** [btcpayserver/btcpayserver#4952] [Technical Support]
Sorry for coming every day with a new issue, but:

If we create an invoice with a metadata property of Array type, it is saved correctly. We can find it returned to the created invoice body and if we get the invoice.

However, the metadata is not visible under the "Additional information" section on the invoice page of btcpay.

I'm reluctant to report this as a bug, because it might be expected.

But I see that you support arrays for example on posData for a Point of Sale invoice, so it could make sense to support it also under metadata: https://docs.btcpayserver.org/Development/InvoiceMetadata/#examples

If this is not the case, it's not a big issue: I will convert it to a comma separated string before creating the invoice and back to an array when I recall the invoice.

Replication:

1. Create an invoice:
```bash
curl --location --request POST 'https://testnet.demo.btcpayserver.org/api/v1/stores/4hEu1jrrC9VKLpLjcVtdLYf1FZTb3bqgUoERboCnGbXE/invoices' \
--header 'Authorization: token <OMITTED>' \
--header 'Content-Type: application/json' \
--data-raw '{
  "currency": "USD",
  "amount": 10,
  "metadata": {
    "buyerEmail": "foo@bar.com",
    "array": [
        "foo",
        "bar"
    ]
  }
}'
```
Response, with the saved array:
```json
{
    "id": "MEPqF2cLE3Tn6wCqeGMCEB",
    "storeId": "4hEu1jrrC9VKLpLjcVtdLYf1FZTb3bqgUoERboCnGbXE",
    "amount": "10",
    "checkoutLink": "https://testnet.demo.btcpayserver.org/i/MEPqF2cLE3Tn6wCqeGMCEB",
    "status": "New",
    "additionalStatus": "None",
    "monitoringExpiration": 1683352786,
    "expirationTime": 1683266386,
    "createdTime": 1683265486,
    "availableStatusesForManualMarking": [
        "Settled",
        "Invalid"
    ],
    "archived": false,
    "type": "Standard",
    "currency": "USD",
    "metadata": {
        "buyerEmail": "foo@bar.com",
        "array": [
            "foo",
            "bar"
        ]
    },
    "checkout": {
        "speedPolicy": "MediumSpeed",
        "paymentMethods": [
            "BTC",
            "BTC-LightningNetwork"
        ],
        "defaultPaymentMethod": null,
        "expirationMinutes": 15,
        "monitoringMinutes": 1440,
        "paymentTolerance": 0.0,
        "redirectURL": null,
        "redirectAutomatically": false,
        "requiresRefundEmail": null,
        "defaultLanguage": null,
        "checkoutType": null
    },
    "receipt": {
        "enabled": null,
        "showQR": null,
        "showPayments": null
    }
}
```
2. Check the invoice page on btcpay, the array metadata is not visible under the "Additional information" section on the invoice page of btcpay.

**Comment by dennisreimann** [comment]
@Kukks Can you take a look?

**Comment by learntheropes** [comment]
Now that I think better, at least for my specific case, this might be even better.

So I can use the array fields as a database without showing them and add better converted strings as the only shown fields.

But since it's posted already, you decide what is better.
